### PR TITLE
Add filter option to get_vts

### PR DESCRIPTION
**Pull request by jjnicola**
It is possible to get a specific collection of VTS with the filter.
Use:
`<get_vts filter='modification_time>20190320'/>`

### CODE REVIEW

File: ospd/misc.py [hunk 1]
```diff
@@ -852,7 +852,7 @@ def main(name, klass):
     cargs = get_common_args(parser)
     logging.getLogger().setLevel(cargs['log_level'])
     wrapper = klass(certfile=cargs['certfile'], keyfile=cargs['keyfile'],
-                    cafile=cargs['cafile'])
+                    cafile=cargs['cafile'], customvtfilter=None)
```
Comment (bjoernricks) [inline ospd/misc.py]: Not required. It's an optional keyword that defaults to None.

```suggestion
                    cafile=cargs['cafile'])
```

File: ospd/vtfilter.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        if not vt_filter:
+            raise RequiredArgument('vt_filter: A valid filter is required.')
+
+        filters = self.parse_filters(vt_filter)
+        if not filters:
+            return None
+
+        _vts_aux = vts.copy()
+        for _element, _oper, _filter_val in filters:
+            for vt_id in _vts_aux.copy():
```
Comment (bjoernricks) [inline ospd/vtfilter.py]: Maybe it is not necessary to create another copy by using? Not sure.
```suggestion
            for vt_id in vts:
```

File: ospd/vtfilter.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        return value
+
+    def format_filter_value(self, element, value):
+        """ Calls the specific function to format value,
+        depending on the given element.
+        """
+        format_func = self.allowed_filter.get(element)
+        return format_func(value)
+
+    def get_filtered_vts_list(self, vts, vt_filter=None):
```
Comment (bjoernricks) [inline ospd/vtfilter.py]: I would not use a optional keyword argument for a required parameter. instead
```suggestion
    def get_filtered_vts_list(self, vts, vt_filter):
```

Besides that the `vts` argument isn't documented.

**Review by bjoernricks — CHANGES_REQUESTED** [review]
Just some minor hints from my side :+1:

File: ospd/vtfilter.py (hunk 2 above)
Comment (jjnicola) [reply ospd/vtfilter.py]: The copy is need since the _vts_aux dictionary is change during the iteration.

Comment (bjoernricks) [reply ospd/vtfilter.py]: Yeah but _vts_aux is already a copy of vts. Therefore IMHO it should be possible to iterate over the original unchanged vts variable.

**Review by bjoernricks — APPROVED** [review]

File: ospd/vtfilter.py (hunk 2 above)
Comment (jjnicola) [reply ospd/vtfilter.py]: At this moment you can pass more than one filter option separated by a semicolon. E.g. : 
`<get_vts filter='modification_time&gt;201903130813;modification_time&lt;201903130815'/>`

The second one must be applied over the result of the first one (logical AND operation). 
If vts is used instead, the logical operation will be an OR, and the result will be the whole vts collection.

**Comment by jjnicola** [comment]
@bjoernricks Thanks a lot for the review!

**Pull request merged**